File: source/arduino/BoardSimulator/I2CLcd.cpp
#include "I2CLcd.h"

I2CLcd::I2CLcd(uint8_t lcd_Addr, uint8_t lcd_cols, uint8_t lcd_rows)
    : lcd(lcd_Addr, lcd_cols, lcd_rows), cols(lcd_cols), rows(lcd_rows) {}

void I2CLcd::begin() {
    lcd.init();
    lcd.backlight();
}

void I2CLcd::print(const char* message) {
    lcd.clear();
    lcd.setCursor(0, 0);
    lcd.print(message);
}

void I2CLcd::clear() {
    lcd.clear();
}

File: source/rpi/BoardSimulator/i2c_lcd.py
# i2c_lcd.py
# I2C LCD simulation for Raspberry Pi

class I2CLcd:
    def __init__(self, address, cols, rows):
        self.address = address
        self.cols = cols
        self.rows = rows
        self.display = ['' for _ in range(rows)]

    def begin(self):
        print(f"LCD initialized at address {hex(self.address)} with size {self.cols}x{self.rows}")

    def print(self, message):
        self.display[0] = message[:self.cols]
        print(f"LCD: {self.display[0]}")

    def clear(self):
        self.display = ['' for _ in range(self.rows)]
        print("LCD cleared")
